components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreAdmin.java

```diff
@@ -362,7 +364,8 @@ public void importCertToStore(String fileName, String certData, String keyStoreN
             if (ks.getCertificateAlias(cert) != null) {
                 // We already have this certificate in the key store - ignore
                 // adding it twice
-                return;
+                throw new SecurityConfigException("Certificate already exists in the keystore with alias: " +
```

Comment:
Won't this be a behavior change?

Reply:
yes, it is. Otherwise, if the certificate is already exist in the Keystore, then the response will be successful(201 Created). Is that ok?

Reply:
In the REST API PoV throwing an exception is fine. 

But previously, based on it implementation this API callee expects to add the certificate only if it does not exist. If we throw an exception won't that behaviour break?

Reply:
As an alternative, I would suggest to keep this method as it is and do the isExists check at the REST API layer.
or instead of a POST at the API level, we can define a PUT.

PUT can create a resource if it does not exists at the path, update if it exists.

Reply:
I have moved that check to the OSGi service https://github.com/wso2/carbon-identity-framework/pull/2529/files#diff-b55e511931238912b7e6ec3ab23ace8bR155
Hope it would be fine